ui/job-view/headerbars/FiltersMenu.jsx

```diff
@@ -9,34 +9,19 @@ export default class FiltersMenu extends React.Component {
 
     this.resultStatuses = thAllResultStatuses.slice();
     this.resultStatuses.splice(thAllResultStatuses.indexOf('runnable'), 1);
-
-    this.toggleResultStatuses = this.toggleResultStatuses.bind(this);
   }
 
   isFilterOn(field) {
-    const { jobFilters } = this.props;
+    const { filterModel } = this.props;
 
     return [
-      ...jobFilters.getResultStatusArray(),
-      ...jobFilters.getClassifiedStateArray(),
+      ...filterModel.urlParams.resultStatus,
+      ...filterModel.urlParams.classifiedState,
     ].includes(field);
   }
 
-  toggleResultStatuses(filterName) {
-    const { jobFilters, recalculateUnclassified } = this.props;
-
-    jobFilters.toggleResultStatuses([filterName]);
-    recalculateUnclassified();
```

Comment:
We now recalculate the unclassified count in the ``ThResultSetStore`` every time the ``history`` listener is triggered in ``JobView``.